hudi-common/src/main/java/org/apache/hudi/common/table/read/InputSplit.java

```diff
@@ -36,33 +39,40 @@
  */
 public class InputSplit {
   private final Option<HoodieBaseFile> baseFileOption;
+  private final Either<Stream<HoodieLogFile>, Iterator<BufferedRecord>> recordsToMerge;
```

Comment:
Revert this to `Option<Iterator<BufferedRecord>>` if the decision is to use keep both the record iterator and a list of log files as members?